src/material/dialog/dialog-content-directives.ts

```diff
@@ -30,13 +30,16 @@ let dialogElementUid = 0;
   host: {
     '(click)': 'dialogRef.close(dialogResult)',
     '[attr.aria-label]': 'ariaLabel || null',
-    'type': 'button', // Prevents accidental form submits.
+    '[attr.type]': 'buttonType',
```

Comment:
I think that we can call the property just `type` since the element itself is a button.